bin/.upgrades/versions/v1/upgradeStorage.js

```diff
@@ -0,0 +1,389 @@
+
+async function moveData (source, target, index, collection, newIndex) {
+  let page = await source.search({
+    index,
+    type: collection,
+    body: { sort: [ '_doc' ] },
+    scroll: '1m',
+    size: 1000
+  });
+
+  const total = page.body.hits.total;
+  let moved = 0;
+
+  while (moved < total) {
+    const bulk = [];
+
+    for (let i = 0; i < page.body.hits.hits.length; i++) {
+      const doc = page.body.hits.hits[i];
+
+      if (doc._source._kuzzle_info) {
+        delete doc._source._kuzzle_info.active;
+        delete doc._source._kuzzle_info.deletedAt;
+      }
+
+      bulk.push({ index: { _index: newIndex, _id: doc._id } });
+      bulk.push(doc._source);
+    }
+
+    await target.bulk({ body: bulk, _source: false });
+
+    moved += page.body.hits.hits.length;
+
+    if (moved < total) {
+      page = await source.scroll({
+        scroll_id: page.body._scroll_id,
+        scroll: '1m'
+      });
+    }
+  }
+
+  return total;
+}
+
+async function upgradeMappings (source, target, index, collection, newIndex) {
+  const
+    mappingsResponse = await source.indices.getMapping({
+      index,
+      type: collection
+    }),
+    mappings = mappingsResponse.body[index].mappings[collection];
+
+  // remove obsolete mapping properties
+  if (mappings.properties && mappings.properties._kuzzle_info) {
+    delete mappings.properties._kuzzle_info.properties.active;
+    delete mappings.properties._kuzzle_info.properties.deletedAt;
+  }
+
+  await target.indices.putMapping({
+    index: newIndex,
+    body: {
+      properties: mappings.properties,
+      dynamic: mappings.dynamic || false,
+      _meta: mappings._meta
+    }
+  });
+}
+
+async function createNewIndex (source, target, index, collection, newIndex) {
+  const exists = await target.indices.exists({ index: newIndex });
+
+  if (exists.body) {
+    await target.indices.delete({ index:newIndex });
+  }
+
+  await target.indices.create({ index: newIndex });
+}
+
+async function upgrade (source, target, index, collection, newIndex) {
+  await createNewIndex(source, target, index, collection, newIndex);
+  await upgradeMappings(source, target, index, collection, newIndex);
+  return await moveData(source, target, index, collection, newIndex);
+}
+
+async function upgradeInternalStorage (context, source, target) {
+  const
+    config = context.config.services.storageEngine.internalIndex,
+    index = `${INTERNAL_PREFIX}${config.name}`,
+    mapconfig = config.collections,
+    collections = {
+      validations: mapconfig.validations,
+      config: mapconfig.config,
+      roles: mapconfig.roles,
+      users: null,
+      profiles: mapconfig.profiles
+    };
+
+  for (const [collection, mappings] of Object.entries(collections)) {
+    const newIndex = getNewIndexName(index, collection);
+    let total;
+
+    if (mappings) {
+      await createNewIndex(source, target, index, collection, newIndex);
+      await target.indices.putMapping({ index: newIndex, body: mappings });
```

Comment:
Why don't you use `upgradeMappings` ?

Reply:
I couldn't find it in Elasticsearch API, did I miss it?

Reply:
I was talking about the method you defined but.. nevermind :expressionless:

Reply:
Ah :)
I don't use it because `upgradeMappings`, as it name suggests, is meant to import a mapping from a Kuzzle v1 source, make it compatible with Kuzzle v2, and then push it to the v2 source.

But we don't want that for internal indexes: v1 mappings should always be considered obsolete, so I take the mappings from the bootstrap configuration directly, and I ignore the v1 mappings.